src/json.c

```diff
@@ -621,6 +622,12 @@ json_decode_item(js_read_T *reader, typval_T *res, int options)
 		/* Check for end of object or array. */
 		if (*p == (top_item->jd_type == JSON_ARRAY ? ']' : '}'))
 		{
+		    if (last_comma && (options & JSON_JS) == 0)
+		    {
+			emsg(_(e_invarg));
```

Comment:
If we do that, then we also need to update the doc a ":help json_decode()" which currently says:
```
                The decoding is permissive:
                - A trailing comma in an array and object is ignored, e.g.
                  "[1, 2, ]" is the same as "[1, 2]".               
```

Reply:
I also that json_decode arrow number-ed key in JSON. Hmm, I prefer strictly.